components/coding/profileV2/AchievementComponent.tsx

```diff
@@ -1,172 +1,161 @@
-import React, { useEffect, useState } from "react";
+import React, { useState } from "react";
 import { PencilAltIcon } from "@heroicons/react/outline";
-import { User } from "../../../graphql/fetchUserProfile";
-import { transformUserBadgeData } from "./AchievementTransformData";
-import { useQuery } from "@apollo/client";
-import { FETCH_CODING_BADGES } from "../../../graphql/coding/userBadges/fetchUserBadges";
+import { useMutation, useQuery } from "@apollo/client";
+import {
+  CodingBadge,
+  FetchBadgeResponse,
+  FETCH_CODING_BADGES,
+  IntroCourseUnit,
+} from "../../../graphql/coding/userBadges/fetchUserBadges";
+import { Button } from "../../ui/Button";
+import {
+  INSERT_USER_CODING_BADGES,
+  DELETE_USER_CODING_BADGES,
+} from "../../../graphql/coding/userBadges/updateUserCodingBadges";
+import UnitBadgeSection from "./achievementcomponenets/UnitBadgeSection";
+import handleOnSaveButtonClick from "./achievementcomponenets/handleSaveOnClick";
 
-export type BadgesSectionProps = {
-  data: User;
+export type AchievementComponentProps = {
+  userId: string;
+  isEditable: boolean;
 };
-export type userCodingBadge = {
-  id: number;
-};
-export type unitProps = {
-  unitTitle: string;
-  codingBadges: codingBadges[];
-};
-export type codingBadges = {
-  id: number;
-  title: string;
-  userCodingBadge: userCodingBadge[];
-  isAwarded: boolean;
-};
-export type unit = unitProps[];
-const AcheivementComponent = ({ user, isEditable = false }) => {
-  const { data } = useQuery(FETCH_CODING_BADGES, {
+
+const AcheivementComponent = ({
```

Comment:
let's get a handle on the naming inconsistency here.. I see the file name is AchievementComponent but you're declaring and exporting the functional component as "AcheivementComponent"... this is getting pretty confusing and is gonna lead to errors down the road